clang/test/dpct/thrust-algo-raw-ptr-noneusm-part2.cu

```diff
@@ -79,3 +82,45 @@ void set_symmetric_difference_by_key() {
   thrust::set_symmetric_difference_by_key(thrust::host, A_keys, A_keys + 7, B_keys, B_keys + 5, A_vals, B_vals, keys_result, vals_result, Compare());
   thrust::set_symmetric_difference_by_key(A_keys, A_keys + 7, B_keys, B_keys + 5, A_vals, B_vals, keys_result, vals_result, Compare());
 }
+
+void swap_ranges() {
+  int v1[2], v2[2];
+
+  // CHECK:  if (dpct::is_device_ptr(v1)) {
+  // CHECK-NEXT:    oneapi::dpl::swap_ranges(oneapi::dpl::execution::make_device_policy(q_ct1), dpct::device_pointer<int>(v1), dpct::device_pointer<int>(v1 + 2), dpct::device_pointer<int>(v2));
+  // CHECK-NEXT:  } else {
+  // CHECK-NEXT:    oneapi::dpl::swap_ranges(oneapi::dpl::execution::seq, v1, v1 + 2, v2);
+  // CHECK-NEXT:  };
+  // CHECK-NEXT:  if (dpct::is_device_ptr(v1)) {
+  // CHECK-NEXT:    oneapi::dpl::swap_ranges(oneapi::dpl::execution::make_device_policy(q_ct1), dpct::device_pointer<int>(v1), dpct::device_pointer<int>(v1 + 2), dpct::device_pointer<int>(v2));
+  // CHECK-NEXT:  } else {
+  // CHECK-NEXT:    oneapi::dpl::swap_ranges(oneapi::dpl::execution::seq, v1, v1 + 2, v2);
+  // CHECK-NEXT:  };
+  thrust::swap_ranges(v1, v1 + 2, v2);
+  thrust::swap_ranges(thrust::host, v1, v1 + 2, v2);
+}
+
+struct Int {
+  __host__ __device__ Int(int x) : val(x) {}
+  int val;
+};
+
+void uninitialized_fill_n() {
+
+  const int N = 137;
+  Int val(46);
+  int data[N];
+
+  // CHECK:  if (dpct::is_device_ptr(data)) {
+  // CHECK-NEXT:    oneapi::dpl::uninitialized_fill_n(oneapi::dpl::execution::make_device_policy(q_ct1), dpct::device_pointer<int>(data), N, val);
+  // CHECK-NEXT:  } else {
+  // CHECK-NEXT:    oneapi::dpl::uninitialized_fill_n(oneapi::dpl::execution::seq, data, N, val);
+  // CHECK-NEXT:  };
+  // CHECK-NEXT:  if (dpct::is_device_ptr(data)) {
+  // CHECK-NEXT:    oneapi::dpl::uninitialized_fill_n(oneapi::dpl::execution::make_device_policy(q_ct1), dpct::device_pointer<int>(data), N, val);
+  // CHECK-NEXT:  } else {
+  // CHECK-NEXT:    oneapi::dpl::uninitialized_fill_n(oneapi::dpl::execution::seq, data, N, val);
+  // CHECK-NEXT:  };
+  thrust::uninitialized_fill_n(data, N, val);
```

Comment:
It will cause compile error: https://godbolt.org/z/9qnrYT855

Reply:
fixed